hubspot/tasks.py

```diff
@@ -68,16 +73,50 @@ def check_hubspot_api_errors():
         defaults={"checked_on": now_in_utc()}
     )
     last_timestamp = hubspot_timestamp(last_check.checked_on)
+
     for error in get_sync_errors():
-        if error.get("errorTimestamp") > last_timestamp:
-            msg = "Hubspot error for {obj_type} id {obj_id}: {details}".format(
-                obj_type=error.get("objectType", "N/A"),
-                obj_id=error.get("integratorObjectId", "N/A"),
-                details=error.get("details", ""),
+        error_timestamp = error.get("errorTimestamp")
+        if error_timestamp > last_timestamp:
+            obj_type = (error.get("objectType", "N/A"),)
+            obj_id = parse_hubspot_id(error.get("integratorObjectId", ""))
+            error_type = error.get("type", "N/A")
+            details = error.get("details", "")
+
+            if (
+                obj_id is not None
+                and "LINE_ITEM" in obj_type
+                and error_type == "INVALID_ASSOCIATION_PROPERTY"
```

Comment:
Need an additional condition here to differentiate between invalid orders (`hs_assoc__deal_id `) or something else, like invalid products (`hs_assoc__product_id`), for example the line resync will never succeed if the error is something like this:

```
{'portalId': 5890463,
  'objectType': 'LINE_ITEM',
  'integratorObjectId': 'xpro-mattbert-4-171',
  'changeOccurredTimestamp': 1560875987887,
  'errorTimestamp': 1560875989036,
  'type': 'INVALID_ASSOCIATION_PROPERTY',
  'details': 'Invalid associations [hs_assoc__product_id: 15]. Do those objects exist?',
  'status': 'OPEN'},
```

A line resync should be attempted only for an invalid order.  An invalid product should be logged and manually fixed.